### ISSUE TITLE
Relaxed Alignments?

### ISSUE THREAD

**Issue by fermat97** [harvardnlp/var-attn#4]
Is _Algorithm 2: relaxed alignment_ included in this repository, since I could only see examples for categorical case. thank you.

**Comment by da03** [comment]
Nope, it's not included here. Relaxed alignment underperforms baseline soft attention (see Table 1 in our paper), and since we need to pretrain with E_p \log p and then finetune to get sensible results, it seems hacky to me. However, if you were interested, it is straightforward to implement using PyTorch's builtin Dirichlet rsample function: https://pytorch.org/docs/stable/_modules/torch/distributions/dirichlet.html#Dirichlet.rsample.
